Add specific error type for "operation already exists" error.

jsafrane opened the pull request:
PersistentVolume controller needs to know why scheduling a new operation has failed - if the operation was already running or some other error happened.

jsafrane commented:
@saad-ali, PTAL

saad-ali commented on pkg/util/goroutinemap/goroutinemap.go:
```diff
@@ -74,3 +74,28 @@ func (grm *goRoutineMap) operationComplete(operationName string) {
+
+var _ error = alreadyExistsError{}
+
+func (err alreadyExistsError) Error() string {
```
nit: This method does not need to be public--it should only be called internally.

jsafrane replied:
Not really, it's part of `error` interface

saad-ali replied:
Ack

saad-ali commented on pkg/util/goroutinemap/goroutinemap.go:
```diff
@@ -74,3 +74,28 @@ func (grm *goRoutineMap) operationComplete(operationName string) {
+	return alreadyExistsError{operationName}
+}
+
+func IsAlreadyExists(err error) bool {
```
nit: add comment on public method

jsafrane replied:
comment added.

jsafrane committed:
Add specific error type for "operation already exists" error.

PersistentVolume controller needs to know why scheduling a new operation has
failed - if the operation was already running or some other error happened.

jsafrane commented:
Rebased

jsafrane commented:
@k8s-bot test this issue: #22655

jsafrane commented:
@k8s-bot test this issue: #25673

jsafrane commented:
@swagiaal, can you please briefly check that I did my rebase right and restore Saad's LGTM?
